fccodegen/data/tests/weblab_model.pyx

```diff
@@ -72,36 +82,92 @@ cdef int _EvaluateRhs(Sundials.realtype var_environment__time, Sundials.N_Vector
     (<Sundials.N_VectorContent_Serial>ydot.content).data[3] = d_dt_n
 
 
-cdef class GeneratedModel_Proto_tmpHzG5bQ(CvodeSolver):
+cdef class GeneratedModel_Proto_tmpHzG5bQ(CvodeSolver): # TODO Generate name
+
+    # The name of the free variable, for use in the ModelWrapperEnvironment
+    # From: fc.simulations.AbstractOdeModel
     cdef public char* freeVariableName
+
+    # The value of the free variable
+    # From: fc.simulations.AbstractOdeModel
     cdef public double freeVariable
+
+    # A mapping from variable name to index within the state variable vector,
+    # for use in the ModelWrapperEnvironment
+    # From: fc.simulations.AbstractOdeModel
     cdef public object stateVarMap
+
+    # A numpy array containing initial values for the state variables
+    # From: fc.simulations.AbstractOdeModel
     cdef public np.ndarray initialState
+
+    # A mapping from parameter name to index within the parameters vector, for
+    # use in the ModelWrapperEnvironment
+    # From: fc.simulations.AbstractOdeModel
     cdef public object parameterMap
+
+    # A numpy array containing model parameter values
+    # From: fc.simulations.AbstractOdeModel
     cdef public np.ndarray parameters
+
+    # An ordered list of the names of the model outputs, as they will be
+    # returned by GetOutputs
+    # From: fc.simulations.AbstractOdeModel
     cdef public object outputNames
 
+    # Mapping from names to saved model states.
+    # From: fc.simulations.AbstractOdeModel
     cdef public object savedStates
+
+    # An instance of fc.utility.environment.ModelWrapperEnvironment
+    # TODO: What's this?
+    # From: fc.simulations.AbstractOdeModel
     cdef public object env
+
+    # True if the solver needs to be reset due to a model change.
+    # TODO This is only ever set via the "environment". Not sure why.
```

Comment:
Because it's the `ModelWrapperEnvironment` that changes model variables from the protocol, and hence needs to flag to CVODE that a change has happened.